xm-commons-function/src/main/java/com/icthh/xm/commons/service/impl/DynamicPermissionCheckServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.icthh.xm.commons.service.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.icthh.xm.commons.config.client.service.TenantConfigService;
+import com.icthh.xm.commons.permission.service.AbstractDynamicPermissionCheckService;
+import com.icthh.xm.commons.permission.service.DynamicPermissionCheckService;
+import com.icthh.xm.commons.permission.service.PermissionCheckService;
+import com.icthh.xm.commons.security.XmAuthenticationContextHolder;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static com.icthh.xm.commons.utils.Constants.FUNCTIONS;
+import static com.icthh.xm.commons.utils.Constants.TENANT_CONFIG_DYNAMIC_CHECK_ENABLED;
+
+@Component
+@ConditionalOnMissingBean(DynamicPermissionCheckService.class)
+public class DynamicPermissionCheckServiceImpl extends AbstractDynamicPermissionCheckService {
+
+    private final TenantConfigService tenantConfigService;
+    private final ObjectMapper mapper;
+
+    public DynamicPermissionCheckServiceImpl(PermissionCheckService permissionCheckService,
+                                             XmAuthenticationContextHolder xmAuthenticationContextHolder,
+                                             TenantConfigService tenantConfigService) {
+        super(permissionCheckService, xmAuthenticationContextHolder);
+        this.tenantConfigService = tenantConfigService;
+        this.mapper = new ObjectMapper();
+    }
+
+
+    public Boolean isDynamicFunctionPermissionEnabled() {
+        return Optional.ofNullable(tenantConfigService.getConfig())
```

Comment:
I would not recommend this, because you will have multiple spec yml files. You have `{spec}.yml` & `/{spec}/*.yml` files and each one will have it's own configuration

Merging this configurations, may not look reliable, imo, because it is not obvious where the configuration came from.

Some global configuration looks better _for now_